src/cwhy/cwhy.py

```diff
@@ -192,34 +192,42 @@ def evaluate_text_prompt(args, prompt, wrap=True, **kwargs):
     return text
 
 
-# Define regular expressions for different compiler error formats
+# Define error patterns with associated information. The numbers
+# correspond to the groups matching file name and line number.
 error_patterns = [
     # C# error message pattern
     ("C#", re.compile(
         r"([a-zA-Z0-9./][^:\r\n]+)\((\d+),(\d+)\): error ([A-Za-z0-9]+): (.*)"
-    )),
+    ), 1, 2),
     # C/C++/Rust error message pattern
     ("C/C++/Rust", re.compile(
         r"([a-zA-Z0-9./][^:->]+):([0-9]+):([0-9]+)"
-    )),
+    ), 1, 2),
     # Java error message pattern
     ("Java", re.compile(
         r"([a-zA-Z0-9./][^:->]+):([0-9]+):"
-    )),
+    ), 1, 2),
     # Python error message pattern
     ("Python", re.compile(
         r'\s*File "(.*?)", line (\d+), in ([^\<].*)'
-    )),
+    ), 1, 2),
     # Go error message pattern
     ("Go", re.compile(
         r"([a-zA-Z0-9./][^:\r\n]+):([0-9]+):([0-9]+): (.*): (.*)"
-    )),
+    ), 1, 2),
     # TypeScript error message pattern
-    re.compile(
+    ("TypeScript", re.compile(
         r"([a-zA-Z0-9./][^:\r\n]+)\((\d+),(\d+)\): error ([A-Za-z0-9]+): (.*)"
-    ),
+    ), 1, 2),
+    # Ruby error message pattern
+    ("Ruby", re.compile(
+        r'"(.*\.rb)", line (\d+)(?:, in `.*\')?: (.*)'
+    ), 1, 2),
+    # PHP error message pattern
+    ("PHP", re.compile(
+        r"PHP (?:Parse|Fatal) error: (.*) in (.*) on line (\d+)"
+    ), 2, 3),
```

Comment:
PHP seems like the only language/regex with different group indices (2/3, others are 1/2). Any reason not to just make this first group non-capturing? Is this future-proofing?

Reply:
More on the future-proofing side. Good point about the non-capturing part, which we could of course do too (as with the actual first group).